lib/runner.js

```diff
@@ -795,7 +795,7 @@ Runner.prototype.run = function (fn) {
   var rootSuite = this.suite;
 
   // If there is an `only` filter
-  if (this.hasOnly) {
+  if (hasOnly(rootSuite)) {
```

Comment:
(careful with nested suites here; `hasOnly()` looks like it only inspects the innards of any given suite, and not if that suite was itself marked as an "only" suite; that information is stored in the aforemention suite's parent! :smile:)